### PR TITLE
Socint 261

### PR DESCRIPTION
**Pull request by KieranWardle**
Replicated valid case filter changes from RHSvc

https://collaborate2.ons.gov.uk/jira/browse/SOCINT-261

### CODE REVIEW

**Review by rpcatling — CHANGES_REQUESTED** [review]
minor things, but ....

Please consider making use of the new TestContainers by adding unit tests to 
 `CaseEventReceiverIT` that check the discard paths when survey or collex not known.  And/Or create a new class `EventFilterIT` to do similar (although the former would be enough I think)

**Review by rpcatling — APPROVED** [review]

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/CaseEventReceiver.java [hunk 1]
```diff
@@ -11,8 +11,6 @@
 import uk.gov.ons.ctp.common.event.model.CaseEvent;
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/CaseEventReceiver.java:11]: Could we rename the 'eventreceiver' classes to accurately reflect what they are receiving please.
When we call RM via REST to get a case, we can optionally request that the case has RM CaseEvent[] populated - CaseEvents are events that RM store against a case.
Suggest 'CaseUpdateEventReceiver' and 'SurveyUpdateEventReceiver'

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    this.collectionExerciseRepository = collectionExerciseRepository;
+  }
+
+  public boolean isValidEvent(String surveyId, String collexId, String caseId, String messageId) {
+
+    log.info("Entering acceptCaseEvent {}, {}", kv("messageId", messageId), kv("caseId", caseId));
+
+    Survey survey = findSurvey(surveyId);
+    boolean acceptedSurveyType = isAcceptedSurveyType(survey);
+    boolean valid = acceptedSurveyType && isKnownCollectionExercise(collexId);
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java:37]: We cannot store the case in the DB without also having the collexid due to the FK constraint, but...
If the case is for a valid _survey_ BUT we don't have the collex YET, does not mean we should throw away the case
We will want to throw the Case back to the subscription / ex manager when it exists, until such time that we get the collex sent to us.
If the survey is OK but we do not have the collex yet, throw a CTPException and let it go back to pub sub

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  private boolean isKnownCollectionExercise(String collexId) {
+    return collectionExerciseRepository.findById(UUID.fromString(collexId)).isPresent();
+  }
+
+  /*
+   * TODO - should we NAK the event/throw exception if we do not recognise
+   * the survey and allow the exception manager quarantine the event or
+   * allow to go to DLQ?
+   */
+  private void logDiscardMessage(
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java:72]: I think this is over complication - just have explicit logging done at the point of discovery

**Review by philwhiles — CHANGES_REQUESTED** [review]
see comments

**Review by philwhiles — APPROVED** [review]

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/CaseEventReceiver.java (hunk 1 above)
Comment (KieranWardle) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/CaseEventReceiver.java:12]: Done

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java (hunk 2 above)
Comment (KieranWardle) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java]: Done plus same for missing survey as discussed

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java (hunk 3 above)
Comment (KieranWardle) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java]: Done

**Review by philwhiles — APPROVED** [review]

**Pull request merged**